Lib/asyncio/locks.py

```diff
@@ -269,17 +269,22 @@ async def wait(self):
                 self._waiters.remove(fut)
 
         finally:
-            # Must reacquire lock even if wait is cancelled
-            cancelled = False
+            # Must reacquire lock even if wait is cancelled.
+            # We only catch CancelledError here, since we don't want any
+            # other (fatal) errors with the future to cause us to spin.
+            err = None
             while True:
                 try:
                     await self.acquire()
                     break
-                except exceptions.CancelledError:
-                    cancelled = True
+                except exceptions.CancelledError as e:
+                    err = e
 
-            if cancelled:
-                raise exceptions.CancelledError
+            if err:
+                try:
+                    raise err  # re-raise same exception instance
```

Comment:
```suggestion
                    raise err  # Re-raise most recent exception instance
```
(If it's possible to be cancelled in this loop, it's possible to be cancelled more than once, so we should consider the possibility of having to choose between multiple cancellation errors. The code wisely chooses the last one caught. Let's reflect that in the comment.)

FWIW, I suspect that the same issue (raising a new cancellation error instead of preserving the instance just caught) exists in many places in the asyncio code base, since the "message" argument to `cancel()` was a late addition to the API (and one that I should have fought harder -- but I was distracted).

Reply:
I think, I guess it is simple, if there is a set of errors raised, to explicitly chain them so that none gets lost.